common/source/docs/common-brushless-escs.rst

```diff
@@ -36,18 +47,97 @@ The DShot ESC protocol's key advantages are:
 - no need to do any ESC throttle range calibration
 - very high protocol frame rates are supported
 
-While DShot is available on many ESCs, those with BLHeli32 capability, offer the most performance and often, additional features such as ESC telemetry. In addition, ArduPilot provides special programming capability for BLHeli32 ESCs via its passthrough feature (See :ref:`common-dshot` section.)
+While DShot is available on many ESCs, those with BLHeli32 setup capability, offer the most performance and often, additional features such as ESC telemetry. In addition, ArduPilot provides special programming capability for BLHeli32 ESCs via its passthrough feature (See :ref:`common-blheli32-passthru` section.)
 
 .. note::
    Recently there is a growing number of proprietary and non-proprietary 16 / 32 bit ESCs with firmware that support DShot and other digital ESC protocols, but not BLHeli_32-specific features like passthrough and telemetry. See your ESC's manual for further detail on supported features.
 
-Where to buy
-============
+Configuration
+=============
+
+Protocol
+--------
+The ESC control protocol is selected by the :ref:`MOT_PWM_TYPE <MOT_PWM_TYPE>` parameter on Copter and Rover, or :ref:`Q_M_PWM_TYPE <Q_M_PWM_TYPE>` on QuadPlanes for copter function motors. Bi-directional DShot is selected by selecting DShot in :ref:`MOT_PWM_TYPE <MOT_PWM_TYPE>` and then selecting the required channels through :ref:`SERVO_BLH_BDMASK <SERVO_BLH_BDMASK>`.
+
+.. note:: The autopilot should be re-booted after changing the protocol type.
+
+ On Plane, other motors use Normal (PWM) protocol. However, in Plane, other motors, like the traditional fixed wing's main motor or Dual Motor Tailsitters (SERVOn_FUNCTION = 70 throttle, 73 throttle left and / or 74 throttle right), can be changed to another protocol using the :ref:`SERVO_BLH_MASK<SERVO_BLH_MASK>` parameter to specify the output number of the motor together with the :ref:`SERVO_BLH_OTYPE<SERVO_BLH_OTYPE>` parameter to select the protocol of these motors independently of that selected for the copter function motors. 
```

Comment:
I think it would be clearer to users if we didn't mixup the configuration across protocols even if that leads to some duplication.  For example, a oneshot users shouldn't need to read this stuff about BLHeli masks.

Reply:
actually for QPlane they do....the only way to designate using a Oneshot ESC on VTOL motors is to use this mask...although the param name is BLH....it may not be a BLHeli ESC....since it allows any non PWM protocol to be chosen for VTOL motors in QPlane...